File: connectors-editors/jira-editor/src/main/scala/com/taskadapter/connector/jira/JiraProjectLoader.scala
package com.taskadapter.connector.jira

import com.google.common.base.Strings
import com.taskadapter.connector.definition.WebConnectorSetup
import com.taskadapter.connector.definition.exceptions.{ConnectorException, ProjectNotSetException, ServerURLNotSetException}
import com.taskadapter.model.GProject
import com.taskadapter.web.callbacks.DataProvider

class JiraProjectLoader(jiraConfig: JiraConfig, setup: WebConnectorSetup) extends DataProvider[GProject] {
  /**
    * Load project info.
    */
  @throws[ConnectorException]
  override def loadData: GProject = {
    if (Strings.isNullOrEmpty(setup.host)) throw new ServerURLNotSetException
    if (jiraConfig.getProjectKey == null || jiraConfig.getProjectKey.isEmpty) throw new ProjectNotSetException
    val project = loadProject(setup, jiraConfig.getProjectKey)
    project
  }

  @throws[ConnectorException]
  private def loadProject(setup: WebConnectorSetup, projectKey: String): GProject = {
    JiraLoaders.validate(setup)
    try {
      val client = JiraConnectionFactory.createClient(setup)
      try {
        val promise = client.getProjectClient.getProject(projectKey)
        val project = promise.claim
        val gProject = new JiraProjectConverter().toGProject(project)
        gProject
      } catch {
        case e: Exception =>
          throw JiraUtils.convertException(e)
      } finally if (client != null) client.close()
    }
  }
}

File: connectors-editors/jira-editor/src/main/java/com/taskadapter/connector/jira/JiraProjectLoader.java
package com.taskadapter.connector.jira;

import com.atlassian.jira.rest.client.api.JiraRestClient;
import com.google.common.base.Strings;
import com.taskadapter.connector.definition.WebConnectorSetup;
import com.taskadapter.connector.definition.exceptions.ConnectorException;
import com.taskadapter.connector.definition.exceptions.ProjectNotSetException;
import com.taskadapter.connector.definition.exceptions.ServerURLNotSetException;
import com.taskadapter.model.GProject;
import com.taskadapter.web.callbacks.DataProvider;

public class JiraProjectLoader implements DataProvider<GProject> {
    private JiraConfig jiraConfig;
    private WebConnectorSetup setup;

    public JiraProjectLoader(JiraConfig jiraConfig, WebConnectorSetup setup) {
        this.jiraConfig = jiraConfig;
        this.setup = setup;
    }

    @Override
    public GProject loadData() throws ConnectorException {
        if (Strings.isNullOrEmpty(setup.host())) {
            throw new ServerURLNotSetException();
        }
        if (jiraConfig.getProjectKey() == null || jiraConfig.getProjectKey().isEmpty()) {
            throw new ProjectNotSetException();
        }
        return loadProject(setup, jiraConfig.getProjectKey());
    }

    private static GProject loadProject(WebConnectorSetup setup, String projectKey) throws ConnectorException {
        JiraLoaders.validate(setup);
        try (JiraRestClient client = JiraConnectionFactory.createClient(setup)) {
            var promise = client.getProjectClient().getProject(projectKey);
            var project = promise.claim();
            return new JiraProjectConverter().toGProject(project);
        } catch (Exception e) {
            throw JiraUtils.convertException(e);
        }
    }
}
